build/typescript-options.js

```diff
@@ -0,0 +1,40 @@
+exports.base = function() {
+  return {
+    "target": "es5",
+    "module": "es2015",
+    "experimentalDecorators": true,
+    "emitDecoratorMetadata": false,
+    "moduleResolution": "node",
+    "stripInternal": true,
+    "preserveConstEnums": true,
+    "declaration": true,
+    "noExternalResolve": false,
+    "removeComments": true,
+    "lib": ["es2015", "dom"],
+    "typescript": require('typescript')
+  }
+}
```

Comment:
Most of the properties listed here are the same as `compilerOptions` in tsconfig.json. So to share the configuration between gulp and IDEs, I've usually imported tsconfig.json compilerOptions and overwritten some properties for gulp. Currently after refactoring lines above could be replaced with following:
```javascript
var tsconfig = require('../tsconfig.json');
var assign = Object.assign || require('object.assign');

exports.base = function() {
  return assign(tsconfig.compilerOptions, {
    "target": "es5",
    "declaration": true,
    "noExternalResolve": false,
    "removeComments": true,
    "typescript": require('typescript')
  });
}
```
and i guess there are some more changes, that could be done: 
1) `base` doesn't need to be exported
2) `"removeComments": true,` could be moved `tsconfig.json` instead